### PR TITLE
adding logout

### PR DESCRIPTION
**Pull request by natashaarmbrust**
- Adds logout
- Fixes bug with highlighting in SettingsPageViewController (only disclosure cells were being selected because of the highlighting bug)

### CODE REVIEW

File: Podcast/UserSettings.swift [hunk 1]
```diff
@@ -81,8 +81,18 @@ class UserSettings: NSObject {
             SettingsSection(id: "profile_settings", items: [
                 SettingsField(id: "username_disclosure", title: "Change Username", type: .disclosure, tapAction: {
                     settingsViewController.navigationController?.pushViewController(UserSettings.changeUsernamePage, animated: true)
+                }),
+            ]),
+            SettingsSection(id: "logout", items: [
+                SettingsField(id: "logout", title: "Logout", type: .button, tapAction: {
+                    guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+                    //TODO: add custom alert view in the future
+                    let alert = UIAlertController(title: "Log out", message: "Are you sure you want to logout?", preferredStyle: .alert)
```
Comment (mindylou) [inline Podcast/UserSettings.swift]: all of the logout related text should be all either "log out" or "logout"

File: Podcast/ActionSheetViewController.swift [hunk 2]
```diff
@@ -3,19 +3,51 @@ import UIKit
 
 class ActionSheetTableViewCell: UITableViewCell {
     
+    let padding: CGFloat = 18
+    var leftPadding: CGFloat = 50
+    var titleLabel: UILabel!
+    var iconImage: UIImageView!
+    
     override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
```
Comment (mindylou) [inline Podcast/ActionSheetViewController.swift:11]: image
the tableviewcells have a slightly different background color than the rest of the actionsheet

File: Podcast/AppDelegate.swift [hunk 3]
```diff
@@ -106,7 +107,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return false
     }
     
-
+    func logout() {
+        googleLoginViewController.logout()
+        window?.rootViewController = loginNavigationController
```
Comment (mindylou) [inline Podcast/AppDelegate.swift:112]: after i logged out and logged back in again it opened to the settings page, it should probably be the feed instead

**Review by mindylou — CHANGES_REQUESTED** [review]
looks good, a couple of small changes!

File: Podcast/UserSettings.swift (hunk 1 above)
Comment (natashaarmbrust) [reply Podcast/UserSettings.swift]: http://whitelightconcepts.com/login-vs-log-in-logout-vs-log-out/
We should probably use log out for both here

File: Podcast/AppDelegate.swift (hunk 3 above)
Comment (natashaarmbrust) [reply Podcast/AppDelegate.swift:112]: fixed

File: Podcast/ActionSheetViewController.swift (hunk 2 above)
Comment (natashaarmbrust) [reply Podcast/ActionSheetViewController.swift:11]: fixed

File: Podcast/ActionSheetViewController.swift [hunk 4]
```diff
@@ -252,11 +261,18 @@ class ActionSheetViewController: UIViewController, UITableViewDataSource, UITabl
         cancelButton.titleLabel?.font = ._14RegularFont()
         cancelButton.addTarget(self, action: #selector(cancelButtonWasPressed), for: .touchUpInside)
         
+        let bottomSeperator = UIView()
```
Comment (mindylou) [inline Podcast/ActionSheetViewController.swift:264]: separator***** HAHA

File: Podcast/SettingsPageViewController.swift [hunk 5]
```diff
@@ -199,11 +198,12 @@ class SettingsPageViewController: ViewController, UITableViewDelegate, UITableVi
             cell.accessoryType = .disclosureIndicator
             break
         case .button:
-            cell.titleLabel.textColor = .blue
+            cell.titleLabel.textColor = .offBlack
```
Comment (doctor3w) [inline Podcast/SettingsPageViewController.swift]: Umm we want Blue, that's how apple does their buttons.

File: Podcast/SettingsPageViewController.swift [hunk 6]
```diff
@@ -225,15 +225,7 @@ class SettingsPageViewController: ViewController, UITableViewDelegate, UITableVi
     func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         return sectionSpacing
     }
-    
-    func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
```
Comment (doctor3w) [inline Podcast/SettingsPageViewController.swift]: Wait don't remove this how does it show selection?

**Review by doctor3w — APPROVED** [review]
I made some changes

**Pull request merged**